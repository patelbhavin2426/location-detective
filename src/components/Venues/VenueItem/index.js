import VenueItem from './venueitem';

export {
    VenueItem
}
