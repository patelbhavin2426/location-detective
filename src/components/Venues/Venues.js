import React, {Component } from 'react'
import PropTypes from 'prop-types';

import {VenueItem} from './VenueItem';

/*
* import {connect } from 'react-redux'
* import { actions } from '../../actions';
*/

class Venues extends Component {

    constructor(props) {
        super(props);
    }

    render(){

        const venues = this.props.venues || [];
        return (<div>

                    {!venues && <div>Search places ..</div> }

                    <ul className="grid space-around">
                        {venues.map( ( item,index) => {

                            const {venue: {name, verified, location}, id} = item;
                            item.index = index;

                            return (
                                    <VenueItem key={index} item={item}/>
                                )
                            })
                        }
                    </ul>
                </div> )

    }

}

/*
const stateToProps = (state) => {

    return {
        venues: state.venues.venues
    }
}


export default connect(stateToProps)(Venues);
*/

Venues.propTypes = {
    venues: PropTypes.array
};

export default Venues;

