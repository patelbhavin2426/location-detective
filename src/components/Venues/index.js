import Venues from './Venues'
import { VenueItem } from './VenueItem'

export {
    Venues,
    VenueItem
}
