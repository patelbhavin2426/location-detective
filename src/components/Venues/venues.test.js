import React from 'react';
import renderer from 'react-test-renderer';

import { shallow, mount } from "enzyme";
import { Provider } from "react-redux";
import configureMockStore from "redux-mock-store";

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Venues } from '../Venues'

let venues = [
    { venue: { name : 'taco bell', verified: true, location: {address : 'El Paso'}} ,  id :'1' },
    { venue: { name : 'taco bell-2', verified: true, location: {address : 'El Paso2'}} , id :'2' }
];

// For Redux
Enzyme.configure({ adapter: new Adapter() });
const mockStore = configureMockStore();
const store = mockStore({
    venues : venues
});


describe("Venue Component :: ", () => {

    it("should render without throwing an error", () => {
        expect(
            renderer.create(
                <Venues venues={venues}/>
            )
        ).not.toBe(null);
    });

    it("should render with Sorry message", () => {

        // Act
        const component = mount(
            <Venues venues={undefined} />
        );

        // Assert
        expect(component).toMatchSnapshot();
    });

    it("should render two items", () => {

        let venues = [
            { venue: { name : 'taco bell', verified: true, location: {address : 'El Paso venue'}} ,  id :'11' },
            { venue: { name : 'taco bell-2', verified: true, location: {address : 'El Paso venue2'}} , id :'22' }
        ];

        // Act
        let component = renderer.create(
            <Venues venues={venues} />
        ).toJSON();
        // snapshot
        expect(component).toMatchSnapshot();

        // Act
        component = mount(
            <Venues venues={venues} />
        );

        // Assert
        expect((component).prop('venues')).not.toEqual(null);
        expect(component.find('li').first().text()).toEqual('taco bell  El Paso venue');
        expect(component.find('li').last().text()).toEqual('taco bell-2  El Paso venue2');
    });

});

