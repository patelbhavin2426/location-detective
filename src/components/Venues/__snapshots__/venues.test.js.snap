// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Venue Component ::  should render two items 1`] = `
<div>
  <ul
    className="grid space-around"
  >
    <li
      className="tile bg-light"
      data-testid="venue11"
    >
      <div
        className="location-name"
      >
        <span
          className="name"
        >
          taco bell
        </span>
         
        <span
          className="fa fa-thumbs-up"
        >
           
        </span>
      </div>
      <div>
        El Paso venue
      </div>
    </li>
    <li
      className="tile bg-light"
      data-testid="venue22"
    >
      <div
        className="location-name"
      >
        <span
          className="name"
        >
          taco bell-2
        </span>
         
        <span
          className="fa fa-thumbs-up"
        >
           
        </span>
      </div>
      <div>
        El Paso venue2
      </div>
    </li>
  </ul>
</div>
`;
